dali/c_api/c_api.h

```diff
@@ -69,6 +75,46 @@ extern "C" {
       int cpu_prefetch_queue_depth,
       int gpu_prefetch_queue_depth);
 
+  /**
+   * @brief Feed the data to ExternalSource as contiguous memory.
+   *
+   * @param pipe_handle Pointer to pipeline handle
+   * @param name Pointer to a null-terminated byte string with the name of the External Source
+   *             to be fed
+   * @param device Device of the supplied memory. Only CPU is supported.
+   * @param data_ptr Pointer to contiguous buffer containing all samples
+   * @param data_type Type of the provided data
+   * @param shape Pointer to an array containing shape of all samples concatenated one after
+   *              another. Should contain batch_size * sample_dim elements.
+   * @param sample_dim The dimensionality of a single sample.
+   * @param layout_str Optional layout provided as a pointer to null-terminated byte string.
+   *                   Can be set to NULL.
+   */
+  DLL_PUBLIC void daliSetExternalInput(daliPipelineHandle* pipe_handle, const char* name,
+                                       device_type_t device, const void* data_ptr,
+                                       dali_data_type_t data_type, const int64_t* shape,
+                                       int sample_dim, const char* layout_str);
```

Comment:
What I don't like about this API here, is that it doesn't forsee strided memory support. Let's say (with GPU external input) user has input data in strided memory. This way he'd need to make one additional copy. And since we actually do a copy there, it might be useful.

I know, strided mem is not supported there yet, but it doesn't look like big  deal, to modify the entry-point copy in Dali

Reply:
It sounds like a great idea for a follow-up task.

Reply:
The support for strides is out of scope for this task (and this PR), we would need to adjust the copying in ExternalSource operator probably to make it work.

Reply:
We would need to add this for C++ API as the Numpy array we accept can have strides as well. So it is definitely bigger than this task PR.